### ISSUE TITLE
Azure Blob useragent enhancement

### ISSUE THREAD

**Issue by xyh1** [duplicati/duplicati#3521]
## Environment info
- **Duplicati version**: Latest
- **Operating system**: All
- **Backend**: Azure Blob

## Description
Please add custom user-agent string to AzureBlob connector. 

Please try hooking the user agent into the operation context sending request event as below to prepend to the existing value of the library:

```
OperationContext.GlobalSendingRequest += (sender, args) =>
    {
        string userAgent = <your user agent> + " " + Microsoft.WindowsAzure.Storage.Shared.Protocol.Constants.HeaderConstants.UserAgent;                
                args.Request.UserAgent = userAgent;
    };
```

Or if using latest SDK, hook into CustomUserAgent property: [OperationContext.CustomUserAgent Property ](https://docs.microsoft.com/dotnet/api/microsoft.windowsazure.storage.operationcontext.customuseragent?view=azure-dotnet#Microsoft_WindowsAzure_Storage_OperationContext_CustomUserAgent)

If possible, please align to the format: APN/1.0 {Product Name}/{Product Version} {Feature Name}/{Feature Version}
e.g. APN/1.0 Duplicati/2.0 AzureBlob/2.0

**Comment by mikaelmello** [comment]
Could tell what are the benefits of doing that?

**Comment by xyh1** [comment]
> 
> 
> Could you tell what are the benefits of doing that?

Yes, I will communicate more over email. I have sent you an email to your address listed on your github profile. 

Thanks.

**Comment by warwickmm** [comment]
I believe this was addressed in pull request #3630.
